lib/insert/insert_duplicate.rb

```diff
@@ -17,9 +17,8 @@ def self.campaign(campaign_id, profile_id)
       raise ParamValidation::ValidationError.new("#{profile_id} is not a valid profile", key: :profile_id)
     end
 
+    dupe = campaign.dup
     Qx.transaction do
-      dupe = campaign.dup
-
```

Comment:
I just realized, I think campaign.dup should happen inside the transaction block. I think if you change line 20 to `dupe=nil`, this should work just fine.